compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -558,6 +558,29 @@ fn transcribe_metavar_expr<'a>(
         span
     };
     match *expr {
+        MetaVarExpr::Concat((lhs, lhs_sign), (rhs, rhs_sign)) => {
+            let string = |ident: Ident, has_sign: bool| {
+                if !has_sign {
+                    return ident.to_string();
+                }
+                let mrni = MacroRulesNormalizedIdent::new(ident);
+                if let Some(nm) = lookup_cur_matched(mrni, interp, &repeats)
+                    && let MatchedNonterminal(nt) = nm
+                    && let Nonterminal::NtIdent(nt_ident, _) = &nt.0
+                {
+                    nt_ident.to_string()
+                } else {
+                    ident.to_string()
+                }
+            };
+            let symbol_span = lhs.span.to(rhs.span);
```

Comment:
Can you add a comment here, describing the hygiene semantics of the resulting span?

Reply:
Ergghhh... Thanks for noticing, spans should use the already defined `visited_span` closure.

Added a new `concat-hygiene.rs` test.